agent/connect.sh

```diff
@@ -1,14 +1,18 @@
 #!/bin/bash
+RUNNER_APPLICATION_DIRECTORY=$HOME/agent/
+source ./utils.sh
+source $RUNNER_APPLICATION_DIRECTORY/.env
 
-user=$(curl -s "http://169.254.169.254/metadata/github_user" | \
-        python3 -c "import sys, json; print(json.load(sys.stdin)['value'])")
-repo=$(curl -s "http://169.254.169.254/metadata/github_repo_name" | \
-        python3 -c "import sys, json; print(json.load(sys.stdin)['value'])")
-vm_name=$(curl -s "http://169.254.169.254/metadata/orka_vm_name" | \
-        python3 -c "import sys, json; print(json.load(sys.stdin)['value'])")
-pat=$(curl -s "http://169.254.169.254/metadata/github_pat" | \
-        python3 -c "import sys, json; print(json.load(sys.stdin)['value'])")
+user=`fetch_encrypted_metadata github_user`
+repo=`fetch_encrypted_metadata github_repo_name`
+github_app_id=`fetch_encrypted_metadata github_app_id`
+github_app_installation_id=`fetch_encrypted_metadata github_app_installation_id`
+github_app_private_key_file=`mktemp`
+sed -E 's/(-+(BEGIN|END) RSA PRIVATE KEY-+) *| +/\1\n/g' <<< $(fetch_encrypted_metadata github_app_private_key) > $github_app_private_key_file
```

Comment:
A comment, to explain what this is doing, might be useful.

Reply:
Removed this as this is moved out of here.